File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>


int count_letters(string w);
int count_words(string t);
int count_sentences(string t);

int main(void)
{
    // Prompt the user for some text
    //string user_text = get_string("Enter text: ");
    string user_text = "This is an example of a sentence."
    // Count the number of letters, words, and sentences in the text
  int num_letters = count_letters(user_text);
  int num_words = count_words(user_text);
  int num_sentences = count_sentences(user_text);


    // Cast into float
    float num_l_f;
    float num_w_f;
    float num_s_f;

    num_l_f = (float) num_letters;
    num_w_f = (float) num_words;
    num_s_f = (float) num_sentences;
    
    // Compute the Coleman-Liau index

  float L = (num_l_f / num_w_f) * 100;
  float S = (num_s_f / num_w_f) * 100;
  
  float index = 0.0588 * L - 0.296 * S - 15.8;
  float index_r = round(index);
  int f_s = (int) index_r;
  if (index_r < 1)
  {
    printf("Before Grade 1\n");
  }
  else if (index_r > 16)
  {
    printf("Grade 16+\n");
  }
  else
  {
    printf("Grade %d\n", f_s); 
  }
}



int count_sentences(string t)
{

  int count = 0;
  int len_t = strlen(t);
  
  for (int i = 0; i < len_t; i++ )
  {
    if (t[i] == '!' || t[i] == '.' || t[i] == '?')
    {
      count++;
    }
  }
  return count;

}


int count_words(string t)
{
  int count = 0;
  int len_t = strlen(t);
  
  for (int i = 0; i < len_t; i++ )
  {
    if (t[i] == ' ')
    {
      count++;
    }
  }
  count += 1;
  return count;

}


int count_letters(string t)
{
  int count = 0;
  int len_t = strlen(t);
  for (int i = 0; i < len_t; i++)
  {
    if (isalpha(t[i]))


    count++;
  }
  return count;
}

File: readability.py
import math


def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count


def count_words(text):
    count = 0
    for char in text:
        if char == " ":
            count += 1
    return count + 1


def count_sentences(text):
    count = 0
    for char in text:
        if char in ["!", ".", "?"]:
            count += 1
    return count


def main():
    # Prompt the user for some text
    user_text = input("Enter text: ")

    # Count the number of letters, words, and sentences in the text
    num_letters = count_letters(user_text)
    num_words = count_words(user_text)
    num_sentences = count_sentences(user_text)

    # Compute the Coleman-Liau index
    L = (num_letters / num_words) * 100
    S = (num_sentences / num_words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    index_r = round(index)
    f_s = int(index_r)

    if index_r < 1:
        print("Before Grade 1")
    elif index_r > 16:
        print("Grade 16+")
    else:
        print(f"Grade {f_s}")


if __name__ == "__main__":
    main()
